kura/org.eclipse.kura.core/OSGI-INF/metatype/org.eclipse.kura.data.DataService.xml

```diff
@@ -98,7 +98,36 @@
             default="0"
             min="0"
             description="Timeouts the in-flight messages congestion condition. The service will force a disconnect attempting to reconnect (0 to disable)."/>
-            
+
+        <AD id="average.publish.rate"
+            name="Average publish rate"
+            type="Integer"
+            cardinality="0"
+            required="true"
+            default="0"
+            min="0"
+            description="The average message publish rate. The unit of measurement is set by the publish rate unit of measurement option. If set to 0 the rate limit is disabled."/>
+        
+        <AD id="average.publish.rate.unit"
+            name="publish rate unit of measurement"
+            type="String"
+            cardinality="0" 
+            required="true"
+            default="messages.second"
+            description="">
+           <Option label="Messages/second" value="messages.second"/>
+           <Option label="Messages/minute" value="message.minute"/>          
+        </AD>
+        
+        <AD id="publish.burst.size"
+            name="Message burst size"
+            type="Long"
+            cardinality="0"
+            required="true"
+            default="0"
+            min="0"
+            description="When the average publish rate is set, this option enables the publishing of messages in bursts at maximum speed. If set to 0 the bursts are disabled."/>
```

Comment:
Just to double check. The broker will not be happy with bursts. Setting this property to 1 disables bursts entirely?